code/controllers/subsystem/lag_switch.dm

```diff
@@ -0,0 +1,133 @@
+/// The subsystem for controlling drastic performance enhancements aimed at reducing server load for a smoother albeit slightly duller gaming experience
+SUBSYSTEM_DEF(lag_switch)
+	name = "Lag Switch"
+	flags = SS_NO_FIRE
+
+	/// If the lag switch measures should attempt to trigger automatically, TRUE if a config value exists
+	var/auto_switch = FALSE
+	/// Amount of connected clients at which the Lag Switch should engage, set via config or admin panel
+	var/trigger_pop = INFINITY - 1337
+	/// List of bools corresponding to code/__DEFINES/lag_switch.dm
+	var/static/list/measures[MEASURES_AMOUNT]
+	/// List of measures that toggle automatically
+	var/list/auto_measures = list(DISABLE_GHOST_ZOOM_TRAY, DISABLE_RUNECHAT, DISABLE_USR_ICON2HTML)
+	/// Timer ID for the automatic veto period
+	var/veto_timer_id
+	/// Cooldown between say verb uses when slowmode is enabled
+	var/slowmode_cooldown = 3 SECONDS
```

Comment:
Slowmode should be like 0.5 seconds imo, 3 seconds affects gameplay way too much